cfpb/consumerfinance.gov
# BrowsePage enhancements

This PR introduces a new optional navigation label for BrowsePages and BrowseFilterablePages (both page types that use our left-hand secondary navigation module). This allows Wagtail editors to provide an alternate label to be used instead of the page title.

This new field is grouped with the existing "exclude siblings" checkbox on the sidebar tab:

<img width="1231" alt="image" src="https://user-images.githubusercontent.com/654645/221917917-f730d990-2d6b-4d04-9a12-e153740874dd.png">

Under the hood I've also done some consolidation of BP and BFP into a single AbstractBrowsePage class that encapsulates the secondary navigation behavior; previously there was some duplication of logic and some logic lived in a v1 util class.

This PR supersedes #7524; thanks to @csebianlander for the idea.

## How to test this PR

The new navigation label should be straightforward to test using any BrowsePage or BrowseFilterablePage.

These changes under the hood should have no impact on user-facing behavior. Using a local production dump, here are some pages I used to validate that:

- BrowsePage, no children, secondary_nav_exclude_sibling_pages False:
http://localhost:8000/enforcement/warning-letters/

- BrowsePage, parent, secondary_nav_exclude_sibling_pages False:
http://localhost:8000/enforcement/payments-harmed-consumers/
- BrowsePage, child, secondary_nav_exclude_sibling_pages False
http://localhost:8000/enforcement/payments-harmed-consumers/payments-by-case/

- BrowsePage, no children, secondary_nav_exclude_sibling_pages True:
http://localhost:8000/ipv6/

- BrowsePage, parent, secondary_nav_exclude_sibling_pages True:
http://localhost:8000/data-research/cfpb-research-conference/2022-cfpb-research-conference/
- BrowsePage, child, secondary_nav_exclude_sibling_pages True:
http://localhost:8000/data-research/cfpb-research-conference/2022-cfpb-research-conference/call-for-papers/

- BrowseFilterablePage, parent, secondary_nav_exclude_sibling_pages True:
http://localhost:8000/about-us/newsroom/
- BrowseFilterablePage, child, secondary_nav_exclude_sibling_pages True:
http://localhost:8000/about-us/newsroom/press-resources/

- BrowseFilterablePage, no browseable children,
secondary_nav_exclude_sibling_pages False:
http://localhost:8000/enforcement/actions/
- BrowseFilterablePage, parent, secondary_nav_exclude_sibling_pages False:
http://localhost:8000/rules-policy/final-rules/
- BrowseFilterablePage, child, secondary_nav_exclude_sibling_pages False:
http://localhost:8000/rules-policy/final-rules/code-federal-regulations/

- PortalSearchPage, parent:
http://localhost:8000/consumer-tools/debt-collection/answers/
- PortalSearchPage, child:
http://localhost:8000/consumer-tools/debt-collection/answers/key-terms/

## Checklist
